src/systems/filecoin_mining/sector/sealing.id

```diff
@@ -35,12 +40,14 @@ type SealVerifyInfo struct {
 // state tree but will be verified in sm.CommitSector. See SealCommitment for
 // data stored on the state tree for each sector.
 type OnChainSealVerifyInfo struct {
-    SealedCID         SealedSectorCID  // CommR
-    SealEpoch         block.ChainEpoch
-    InteractiveEpoch  block.ChainEpoch
-    Proof             SealProof
-    DealIDs           deal.DealIDs
+    SealedCID             SealedSectorCID  // CommR
+    SealEpoch             block.ChainEpoch
+    InteractiveEpoch      block.ChainEpoch
+    Proof                 SealProof
+    DealIDs               deal.DealIDs
     SectorNumber
+
+    IsValidAtSealEpoch()  bool
 }
```

Comment:
@porcuquine Do you not need the randomness on chain for seal?

Reply:
Did I remove it? I think both randomness values are derived from the corresponding epochs when needed.

Reply:
Yes. We chose to explicitly include randomness in the ePoSt case given the randomness is a VRF output. Note that there is a potential space saving step we hadn't considered if we choose to do the surprisePoSt without the VRF prior, in which case we can just use epoch number for surprisePoSt onchain output. The downside being it means the two posts are now meaningfully different.

Reply:
Got it. Let's fix later, since it's unrelated to this change set we need for other reasons?